src/ApiService/ApiService/ApiService.csproj

```diff
@@ -34,6 +34,7 @@
     <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.17.0" />
     <PackageReference Include="System.Linq.Async" Version="6.0.1" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.SignalRService" Version="1.7.0" />
+    <PackageReference Include="TaskTupleAwaiter" Version="2.0.0" />
```

Comment:
I didn't see this used anywhere

Reply:
It adds into the `System.Threading.Tasks` namespace; allows the usage as seen in `OnWorkerEventDone` where you can do:

```csharp
var (task, node) = await (
    _context.TaskOperations.GetByTaskId(done.TaskId),
    _context.NodeOperations.GetByMachineId(machineId));
```

this will run the tasks concurrently so that they can hit Azure Storage in parallel.